### ISSUE TITLE
[member-ordering] Properties with literal keys are wrongly reported in alphabetical order

### ISSUE THREAD

**Issue by avaly** [typescript-eslint/typescript-eslint#4051]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/member-ordering": [2, {
      "default": { "order": "alphabetically" }
    }]
  }
}
```

```TS
interface TestMemberOrder {
  title: string;
  'xyz.foo': string;
}
```

**Expected Result**

The code is valid.

**Actual Result**

The code throws an error due to the literal key of the last property:

```
  3:3  error  Member "xyz.foo" should be declared before member title  @typescript-eslint/member-ordering
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `5.1.0` + `canary` |
| `@typescript-eslint/parser`        | `5.1.0` |
| `TypeScript`                       | `4.4.4` |
| `ESLint`                           | `8.1.0` |
| `node`                             | `14.18.1` |
